osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -69,7 +69,7 @@ public abstract class ScoreProcessor
         /// <summary>
         /// Whether the score is in a failed state.
         /// </summary>
-        public virtual bool HasFailed => Health.Value == Health.MinValue;
+        public virtual bool HasFailed => alreadyFailed || Health.Value == Health.MinValue && Failed?.Invoke() != false;

```

Comment:
putting this invoke call in a getter property isn't good. the `Failed?.Invoke` part needs to remain in `UpdateFailed`